src/wtforms/fields/core.py

```diff
@@ -1203,8 +1209,8 @@ def _add_entry(self, formdata=None, data=unset_value, index=None):
         if index is None:
             index = self.last_index + 1
         self.last_index = index
-        name = "%s-%d" % (self.short_name, index)
-        id = "%s-%d" % (self.id, index)
+        name = self.short_name + self._separator + str(index)
```

Comment:
This appears to be confusing in the interpolation and operation of concatenation. Can you make this more explicit in the way it was?

Reply:
Sure thing, but may I ask why interpolation is more explicit in this case? 
Once the separator is a string variable as well the interpolation is `"%s%s%d"` which isn't particularly explicit either, I would argue.

Reply:
A couple things:

- + operator is used on floats, ints, strings etc
- the type of `name` isn't explicit with your change, it is clearly a string when doing `"%s-%d"`

Reply:
I just saw that the minimum python version is 3.6 so we can use f-strings.
Would you agree that `f"{self.short_name}{self._separator}{index}"` is the nicest syntax?

Reply:
Seems ok to me.